File: String_Functions/Substrings/Subs.py
def make_sub(s):
    n = len(s)
    subs = []
    for i in range(n):
        for j in range(i + 1, n + 1):
            subs.append(s[i:j])
    return subs

def main():
    a = "abcd"
    substrings = make_sub(a)
    for s in substrings:
        print(s)

if __name__ == "__main__":
    main()

File: String_Functions/Substrings/Subs.cpp
#include <iostream>
#include <vector>
using namespace std;


vector<string> makeSub(const string& s) {
    int n = s.size();
    vector<string> subs;
    for(int i=0;i<n;i++) {
        for(int j=i+1;j<=n;j++) {
            subs.push_back(s.substr(i,j-i));
        }
    }
    return subs;
}

int main() {

    string a = "abcd";
    vector<string> Substrings = makeSub(a);
    for(string s:Substrings)  cout<<s<<endl;

    return 0;
}